io-engine/src/core/volume_snapshot.rs

```diff
@@ -0,0 +1,46 @@
+use super::CoreError;
+use crate::{
+    lvs::{Lvol, LvolSnapshot},
+    subsys::NvmfReq,
+};
+use async_trait::async_trait;
+use spdk_rs::libspdk::spdk_bdev_io;
+
+///  Traits gives the common snapshot/clone interface for Local/Remote Lvol.
+#[async_trait(?Send)]
+pub trait VolumeSnapshotOps {
+    /// Create Snapshot Common API.
+    async fn create_snapshot(&self) -> Result<u64, CoreError>;
+
+    /// Return the list of Snapshots for Local/remote Device.
+    fn list_snapshot(&self, source_vol: &str) -> Vec<LvolSnapshot>;
+}
+pub trait VolumeSnapshotParams {
+    /// Get Transaction Id of the Snapshot Create.
+    fn txn_id(&self) -> Option<String>;
+
+    /// Get Entity Id of the Snapshot.
+    fn entity_id(&self) -> Option<String>;
+
+    /// Get Parent Id of the Snapshot.
+    fn parent(&self) -> Option<String>;
+
+    /// Get Snapshot Name.
+    fn snap_name(&self) -> Option<String>;
+
+    /// Get Lvol.
+    fn lvol(&self) -> &Lvol;
```

Comment:
What kind of volume are we obtaining ? The parent of the snapshot ?

Reply:
How come we can get the parent directly, but not the parent id?

Reply:
On another note, should this be in this trait? Seems we're constraining this snapshot to be a blobstor snapshot?
Maybe we could have another trait, for example:
pub trait LvolSnapshotParams: VolumeSnapshotParams { {
   /// Get Lvol.
    fn lvol(&self) -> &Lvol;
}